students/tammyd/lesson06/test_mailroom_pt4.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+
+#Unit testing
+
+import os
+import mailroom_pt4 as mailroom
+
+mailroom.donor_db = mailroom.donor_db
+
+#Lots of failures but I did my best to show my work.
+#Is it necessary to test every single function?
+
+
+
+menu_dict = mailroom.main_menu()
+menu_selection = menu_dict("T")
+
+
+def test_donor_database():
+    assert return donor_database == donor_db
+    
+
+#what is causing this error?: 
+#"NameError: name 'menu_dict' is not defined"
+def test_main_menu():
+    menu_selection = mailroom.main_menu("T")
+    assert menu_dict[0] == "T"
+    assert letters_to_everyone in menu_dict
+
+
+#what is causing this failure?: 
+#"TypeError: 'NoneType' object is not subscriptable"
+def test_find_donor():
+   donor = mailroom.find_donor("queeN bey")
+   assert donor[0] == "Queen Bey" 
+#   donor = mailroom.find_donor("mandy MOOre")
+#   assert donor[0] == "Mandy Moore"
+
+#Why does this pass even when I test "qenn bey" instead of "Queen Bey"?
+def test_find_donor_not():
+    "test one that's not there"
+    donor = mailroom.find_donor("qenn bey")
+    assert donor is None
```

Comment:
This will pass if None is None. And since find_donor won't find "qenn bey" it'll return None. Then you'll assert donor, which means none is none, so that'll pass. It won't pass if you can find the donor. It's exactly what it looks like it should do.